test/new-e2e/utils/infra/stack_manager.go

```diff
@@ -60,8 +60,24 @@ func newStackManager(ctx context.Context) (*StackManager, error) {
 	}, nil
 }
 
+func getNewStack(ctx context.Context, name, project string, deployFunc pulumi.RunFunc, create bool) (auto.Stack, error) {
+	var newStack auto.Stack
+	var err error
+
+	if create {
+		newStack, err = auto.UpsertStackInlineSource(ctx, name, projectName, deployFunc)
```

Comment:
Isn't it the definition of `Upsert` to create if not present and select stack otherwise?

Reply:
Yes, but we also wanted to allow for the case where we select if present, and fail otherwise.